coordinator/src/orderbook/routes.rs

```diff
@@ -92,17 +93,37 @@ pub async fn post_order(
         update_pricefeed(OrderbookMsg::NewOrder(order.clone()), sender);
     }
 
-    let all_orders =
-        orders::all_by_direction_and_type(&mut conn, order.direction.opposite(), OrderType::Limit)
-            .map_err(|e| {
-                AppError::InternalServerError(format!("Failed to load all orders: {e:#}"))
-            })?;
+    let all_orders = orders::all_by_direction_and_type(
+        &mut conn,
+        order.direction.opposite(),
+        OrderType::Limit,
+        false,
+    )
+    .map_err(|e| AppError::InternalServerError(format!("Failed to load all orders: {e:#}")))?;
     let matched_orders = match_order(order.clone(), all_orders)
         .map_err(|e| AppError::InternalServerError(format!("Failed to match order: {e:#}")))?;
 
     let authenticated_users = state.authenticated_users.lock().await;
     if let Some(matched_orders) = matched_orders {
+        let mut orders_to_set_taken = vec![matched_orders.taker_matches.filled_with.order_id];
+        let mut order_ids = matched_orders
+            .taker_matches
+            .filled_with
+            .matches
+            .iter()
+            .map(|m| m.order_id)
+            .collect();
+
+        orders_to_set_taken.append(&mut order_ids);
+
         notify_traders(matched_orders, authenticated_users.clone()).await;
+
+        for order_id in orders_to_set_taken {
+            if let Err(err) = db::orders::update(&mut conn, order_id, true) {
```

Comment:
nit: the method name could be more clear to what this is doing. I would have expected an update method to update the whole order and not just the taken flag. How about `db::orders::taken(&mut conn, order_id, true)`?

Reply:
done